plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -835,84 +920,83 @@ fu_genesys_usbhub_device_setup(FuDevice *device, GError **error)
 		self->flash_sector_size = sector_size;
 
 	/* setup firmware parameters */
-	switch (self->isp_model) {
-	case ISP_MODEL_HUB_GL3523:
+	switch (self->chip.model) {
+	case ISP_MODEL_HUB_GL3523: {
+		gint var = 0;
+		guint8 bonding = 0;
 		self->fw_bank_addr[0] = 0x0000;
 		self->fw_bank_addr[1] = 0x8000;
 		self->extend_size = GL3523_PUBLIC_KEY_LEN + GL3523_SIG_LEN;
-		if (self->isp_revision == 50) {
+
+		if (self->chip.revision == 50) {
 			self->fw_data_total_count = 0x8000;
-			if (!fu_genesys_usbhub_device_get_fw_size(self, 0, error))
-				return FALSE;
+			self->fw_key_addr[0] = 0x7D00;
+			self->fw_key_addr[1] = 0xFD00;
 		} else {
 			self->fw_data_total_count = 0x6000;
-			self->code_size = self->fw_data_total_count;
+			self->fw_key_addr[0] = 0x6000;
+			self->fw_key_addr[1] = 0xE000;
+			self->code_size[0] = self->code_size[1] = self->fw_data_total_count;
 		}
-		break;
-	case ISP_MODEL_HUB_GL3590:
-		if (!fu_genesys_usbhub_device_get_fw_size(self, 0, error))
+
+		var = fu_genesys_tsdigit_value((gchar)self->dynamic_ts.bonding);
+		if (var == -1) {
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INTERNAL,
+				    "GL3523 bonding value(%hhX) is wrong",
```

Comment:
`"wrong bonding value %02x"`,

Note: I have not seen `%hh` before.

Reply:
`0x%02x` please -- then we know it's hex at a glance

Reply:
yes forgot the `0x` prefix